source/Tutorials/Advanced/Security/Deployment-Guidelines.rst

```diff
@@ -0,0 +1,214 @@
+In fact, two different CA's can be selected for identity and permissions, per enclave.
+Those CA artifacts are stored inside ``private/`` and ``public/`` sub-directories of a `Keystore <https://design.ros2.org/articles/ros2_security_enclaves.html>`_ with the following folder structure:
+
+.. code-block:: text
+
+  keystore
+  ├── enclaves
+  │   └── ...
+  │       └── ...
+  ├── private
+  │   └── ...
+  └── public
+      └── ...
+
+A good practice for the creation and usage of a certain Certificate Authority on a typical deployment over a production system, is to:
+
+  (1) Create it within the organization system intended for internal use only.
+  (2) Generate/modify desired enclaves.
+  (3) Ship ``public/`` alongside with corresponding ``enclaves/`` into the different remote production systems during setup, depending on the communication needs (not all remote systems may need to interact with each other).
+  (4) Keep and protect ``private/`` keys and/or certification requests in the organization.
+
+It is important to note that if ``private/`` files are lost, it won't be possible to change access permissions, adding or modifying security profiles anymore.
+
+In addition, further practices may be taken into consideration:
+
+* Granting read-only permissions to the ``enclaves/`` directory contents.
+* If a PKCS#11 compliant URI is given for generating enclave's private keys, a `Hardware Security Module (HSM) <https://en.wikipedia.org/wiki/Hardware_security_module>`_ could be used to store them.
+
+The following table depicts a summary of the previous statements relating the Keystore directory with the Recommended location:
+
++------------------------+--------------+---------------+---------------------+
+| Directory / Location   | Organization | Remote System | Material Sensitivity|
++========================+==============+===============+=====================+
+| public                 |       ✓      |       ✓       |         Low         |
++------------------------+--------------+---------------+---------------------+
+| private                |       ✓      |       ✕       |         High        |
++------------------------+--------------+---------------+---------------------+
+| enclaves               |       ✓      |       ✓       |        Medium       |
++------------------------+--------------+---------------+---------------------+
+
+
+Building a deployment scenario
+------------------------------
+
+To illustrate a simple deployment scenario, a docker image will be built intented to be the remote production system actor.
+In this example, localhost will serve as the organization's system.
+To test security capabilities, a secure listener will be launched on the remote system, whereas a secure talker will be launched on the localhost.
+Let's start by creating a workspace tree with the following sub-folders:
+
+.. code-block:: bash
+
+  mkdir -p ~/deploy_gd_tutorial/organization_system
```

Comment:
I do not have a local installation of ROS 2, so I used `ros:rolling` image instead, sharing a volume for the created files. Once the dirs and files where created, I built the docker image locally given the `keystore` dir from the shared volume as context (I had to change the owner to my user for this). Everything went well and means I can create all this organization stuff without installing ROS 2
 and everything went very good

Reply:
Nice to know ! 
For the moment, I will leave it as it is and as have re-checked the tutorial as it is. ( i have not got a local ros2 installation neither, but built from sources  :smiley: )